src/main/cliapp/src/containers/conditionRelationPage/ConditionRelationTable.js

```diff
@@ -17,326 +17,500 @@ import {ErrorMessageComponent} from "../../components/ErrorMessageComponent";
-      >
-          <Column field='rowEditor' rowEditor style={{maxWidth: '7rem', minWidth: '7rem'}}
-            headerStyle={{ width: '7rem', position: 'sticky' }} bodyStyle={{ textAlign: 'center' }} frozen headerClassName='surface-0'/>
-        {columnList}
-      </DataTable>
-    </div>
-  )
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+   }, [tableState, isEnabled, columnWidths]);
+
+   const colReorderHandler = (event) => {
+      let _columnNames = [...tableState.selectedColumnNames];
+      _columnNames = reorderArray(_columnNames, event.dragIndex, event.dropIndex);
+      setSelectedColumnNames(_columnNames);
+   };
+
+   const handleColumnResizeEnd = (event) => {
+      const currentWidth = event.element.clientWidth;
+      const delta = event.delta;
+      const newWidth = Math.floor(((currentWidth + delta) / window.innerWidth) * 100);
+      const field = event.column.props.field;
+
+      const _columnWidths = {...columnWidths};
+
+      _columnWidths[field] = newWidth;
+      setColumnWidths(_columnWidths);
+   };
+
+   return (
+      <div className="card">
+         <Toast ref={toast_topleft} position="top-left"/>
+         <Toast ref={toast_topright} position="top-right"/>
+         <Messages ref={errorMessage}/>
+         <DataTable value={conditionRelations} header={header} reorderableColumns={isEnabled}
+                    tableClassName='p-datatable-md' scrollable scrollDirection="horizontal" tableStyle={{width: '200%'}} scrollHeight="62vh"
+                    editMode="row" onRowEditInit={onRowEditInit} onRowEditCancel={onRowEditCancel} onRowEditSave={(props) => onRowEditSave(props)}
+                    onColReorder={colReorderHandler}
+                    editingRows={editingRows} onRowEditChange={onRowEditChange}
+                    ref={dataTable}
+                    filterDisplay="row"
+                    onColReorder={colReorderHandler}
+                    sortMode="multiple" removableSort onSort={onSort} multiSortMeta={tableState.multiSortMeta}
+                    first={tableState.first}
+                    dataKey="id" resizableColumns columnResizeMode="expand" showGridlines onColumnResizeEnd={handleColumnResizeEnd}
+                    paginator totalRecords={totalRecords} onPage={onLazyLoad} lazy
+                    paginatorTemplate="CurrentPageReport FirstPageLink PrevPageLink PageLinks NextPageLink LastPageLink RowsPerPageDropdown"
+                    currentPageReportTemplate="Showing {first} to {last} of {totalRecords}" rows={tableState.rows} rowsPerPageOptions={[10, 20, 50, 100, 250, 1000]}
+         >
+            <Column field='rowEditor' rowEditor style={{maxWidth: '7rem', minWidth: '7rem'}}
+                    headerStyle={{width: '7rem', position: 'sticky'}} bodyStyle={{textAlign: 'center'}} frozen headerClassName='surface-0'/>
+            {columnList}
+         </DataTable>
+      </div>
+   )
```

Comment:
This whole file has changed due to tab / space issue. The repo is set for 2 spaces for indent for JS and 4 spaces for Java. We need to switch this over to tabs and then every developer can choose whatever tabstop they would like.